File: online-judge-solutions/Codeforces/1714B.cpp
//https://codeforces.com/problemset/problem/1714/B
//Lang		:	GNU C++14
//Time		:	312 ms
//Memory	:	7200 KB
#include<iostream>
#include<vector>
#include<set>
#include<algorithm>
using namespace std;
int main(){
	int t,n;
	cin>>t;
	while(t--){
		cin>>n;
		vector<int> v(n);
		for(int i=0;i<n;i++){
			cin>>v[i];
		}
		bool b=true;
		set<int> s;
		for(int i=n-1;i>=0;i--){
			if(s.count(v[i])){
				cout<<i+1<<endl;
				b=false;
				break;
			}
			s.insert(v[i]);
		}
		if(b) cout<<0<<endl;
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1714B.py
#https://codeforces.com/problemset/problem/1714/B
#Lang		:	Python 3.8
#Time		:	265 ms
#Memory         :	13500 KB
t=int(input())
for x in range(t):
    n=int(input())
    v=list(map(int,input().split()))[:n]
    b,s=True,set()
    for i in range(n-1,-1,-1):
        s2={v[i]}
        if(s2.issubset(s)):
            print(i+1)
            b=False
            break
        s.add(v[i])
    if(b):print(0)
